Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs

```diff
@@ -31,6 +31,46 @@ public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
         {
             dequeuedTask = await _sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
 
+            if (string.IsNullOrEmpty(dequeuedTask.ExistentBusinessPartner.FederalTaxID))
+            {
+                return await CreateBusinessPartner(dequeuedTask);
+            }
+            else
+            {
+                return await UpdateBusinessPartner(dequeuedTask);
+            }
+        }
+
+        private async Task<SapTaskResult> CreateBusinessPartner(SapTask dequeuedTask)
+        {
+            var message = new HttpRequestMessage()
+            {
+                RequestUri = new Uri($"{_sapConfig.BaseServerUrl}BusinessPartners/"),
+                Content = new StringContent(JsonConvert.SerializeObject(dequeuedTask.BusinessPartner),
+                    Encoding.UTF8,
+                    "application/json"),
+                Method = HttpMethod.Post
+            };
+
+            var cookies = await _sapTaskHandler.StartSession();
+            message.Headers.Add("Cookie", cookies.B1Session);
+            message.Headers.Add("Cookie", cookies.RouteId);
+
+            var client = _httpClientFactory.CreateClient();
+            var sapResponse = await client.SendAsync(message);
```

Comment:
What do you think use the next  SendMessage method, for example
`var sapResponse = await SendMessage(dequeuedTask, HttpMethod.Post, new 
Uri(${_sapConfig.BaseServerUrl}BusinessPartners/"));` 
you have HttpRequestMessage and StartSession duplicate in this class

Reply:
Good catch, I going to refactor this.